capsules/extra/src/ieee802154/framer.rs

```diff
@@ -468,13 +470,18 @@ impl<'a, M: Mac<'a>, A: AES128CCM<'a>> Framer<'a, M, A> {
     }
 
     /// IEEE 802.15.4-2015, 9.2.3, incoming frame security procedure
-    fn incoming_frame_security(&self, buf: &'static mut [u8], frame_len: usize) -> RxState {
+    fn incoming_frame_security(
+        &self,
+        buf: &'static mut [u8],
+        frame_len: usize,
+        lqi: u8,
+    ) -> RxState {
         // Try to decode the MAC header. Three possible results can occur:
         // 1) The frame should be dropped and the buffer returned to the radio
         // 2) The frame is unsecured. We immediately expose the frame to the
         //    user and queue the buffer for returning to the radio.
         // 3) The frame needs to be unsecured.
-        let result = Header::decode(&buf[radio::PSDU_OFFSET..], false)
+        let result = Header::decode(&buf[radio::PSDU_OFFSET..(radio::PSDU_OFFSET+radio::MAX_FRAME_SIZE)], false)
```

Comment:
Oh I meant to ask about these changes as well. These seem unrelated to LQI. Are they necessary?

Reply:
This gets at a major pain point in the 15.4 network stack...

The short answer is that the buffer given to the radio was increased by one byte (on nrf52840 and rf233 the lqi is placed by the radio after the packet). 

Ideally, we would then only pass a buffer of the size of the radio packet (as done previously). However, we must pass the whole buffer since the buffer is given back to the radio after being passed up through the network stack (this is a major drawback of our current design). What this in reality means is that to keep a 130 byte buffer (size needed for psdu + packet + lqi), we have to pass a buffer of this size up the entire stack. Passing a slice results in the buffer "shrinking" each time a packet is received as the radio's buffer is then replaced with the now truncated slice. This is not a unique problem and is also caused friction with the PSDU offset PR a few weeks ago.

So the reason we have to now set the upper bound on the range is so that we don't pass the lqi as part of the packet to layers higher than the framer (since the buffer at the framer includes the lqi too). I'll go ahead and add a brief comment mentioning this.